lib/Conversion/ImportVerilog/Statements.cpp

```diff
@@ -463,6 +473,31 @@ struct StmtVisitor {
     return success();
   }
 
+  // Handle immediate assertion statements.
+  LogicalResult visit(const slang::ast::ImmediateAssertionStatement &stmt) {
+    auto cond = context.convertRvalueExpression(stmt.cond);
+    cond = context.convertToBool(cond);
+    if (!cond)
+      return failure();
+
+    auto defer = convertDeferAssert(stmt.isDeferred, stmt.isFinal);
```

Comment:
You could inline `convertDeferAssert` here and make it switch to different assert flavors conservatively:
```c++
auto defer = moore::DeferAssert::Immediate;
if (stmt.isFinal)
  defer = moore::DeferAssert::Final;
else if (stmt.isDeferred)
  defer = moore::DeferAssert::Observed;
```

Reply:
Yeah, I need to inline `convertDeferAssert`. I did it.